agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -1106,6 +1152,52 @@ bool monitor_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
         mon_wlan_hal->sta_link_measurements_11k_request(mac_str);
         break;
     }
+    case beerocks_message::ACTION_MONITOR_CHANNEL_SCAN_TRIGGER_SCAN_REQUEST: {
+        auto request =
+            beerocks_header
+                ->addClass<beerocks_message::cACTION_MONITOR_CHANNEL_SCAN_TRIGGER_SCAN_REQUEST>();
+        if (!request) {
+            LOG(ERROR) << "addClass cACTION_MONITOR_CHANNEL_SCAN_TRIGGER_SCAN_REQUEST failed";
+            return false;
+        }
+
+        auto radio_mac         = request->scan_params().radio_mac;
+        auto dwell_time_ms     = request->scan_params().dwell_time_ms;
+        auto channel_pool      = request->scan_params().channel_pool;
+        auto channel_pool_size = request->scan_params().channel_pool_size;
+        auto channel_pool_vector =
+            std::vector<unsigned int>(channel_pool, channel_pool + channel_pool_size);
+        std::string channels;
+        for (int8_t index = 0; index < channel_pool_size; index++) {
```

Comment:
`channel_pool_size` is a `uint8_t`, `index` should probably also be a uint8_t.

Unrelated to this PR:
channel_pool_size should also probably be of a size type since it's a size.

Reply:
Changed the for loop to
`for (int index = 0; index < int(channel_pool_size); index++) {`
that way it is not dependent on what type is the `channel_pool_size`

And regarding the `channel_pool_size`, it needs to be a basic type for the CMDUs